linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -100,68 +93,30 @@ final class Emitter private (config: CommonPhaseConfig,
         internalOptions.withTrackAllGlobalRefs(trackAllGlobalRefs))
   }
 
-  def emitAll(unit: LinkingUnit, builder: JSLineBuilder,
-      logger: Logger): Unit = {
-    emitInternal(unit, builder, logger) {
-      val topLevelVars = topLevelVarDeclarations(unit)
-      if (topLevelVars.nonEmpty) {
-        val kw = if (esFeatures.useECMAScript2015) "let " else "var "
-        builder.addLine(topLevelVars.mkString(kw, ", ", ";"))
-      }
-
-      if (needsIIFEWrapper)
-        builder.addLine("(function(){")
```

Comment:
Hmm... So I put this in `Emitter` because IMHO it logically belongs there. GCC is the odd one out, because we do not have an easy way to prevent it from renaming symbols that we define (defining an extern will cause a duplicate definition).

I have been toying with the idea of adding calls to `goog.ExportSymbol` (GCC primitive IIUC) to the code we sent to GCC instead of using externs which should fix this. If we do this, module construction is entirely the responsibility of the `Emitter`.

> It would avoid the increase in code size, since the inside would not be indented for no reason.

IMHO a weak point when compared to making the separation cleaner (and the additional typing/safety of always having trees), but indeed the separation is not much cleaner right now.

If you like, I can fiddle with the `goog.ExportSymbol` idea first (maybe I can also clean up the properties mess :S).

Reply:
It's unlikely that we can change the way we talk to GCC. We have to pass it the code outside any top-level IIFE, otherwise it does not optimize as much. So I'm not sure that exploring `goog.ExportSymbol` is going to significantly improve the situation. It might remove the need to provide specific externs, but the Closure backend would still have to deal with the declaration of top-level exports, the IIFE and the `"use strict"` directive.

Reply:
OK. In this case its indeed better to solely use `emitBody` for both.